nrich-excel/src/test/java/net/croz/nrich/excel/service/DefaultExcelReportServiceTest.java

```diff
@@ -48,37 +48,51 @@ void shouldCreateExcelReport() {
         assertThat(sheet).isNotNull();
 
         assertThat(sheet.getRow(0).getCell(0)).isNotNull();
-        assertThat(getRowCellValueList(sheet.getRow(TEMPLATE_DATA_FIRST_ROW_INDEX))).containsExactly(rowData[0]);
+        assertThat(getRowCellValueList(sheet.getRow(TEMPLATE_DATA_FIRST_ROW_INDEX))).containsExactly(DEFAULT_ROW_DATA[0]);
     }
 
     @Test
     void shouldThrowExceptionOnInvalidBatchSize() {
         // given
         File file = createFileInTemporaryDirectory();
-        Object[][] rowData = new Object[][] { { 1.1, "value" } };
-        MultiRowDataProvider multiRowDataProvider = (start, limit) -> start == 0 ? rowData : null;
-
-        CreateExcelReportRequest request = CreateExcelReportRequest.builder().multiRowDataProvider(multiRowDataProvider).batchSize(-1).outputFile(file).templatePath("classpath:excel/template.xlsx").firstRowIndex(TEMPLATE_DATA_FIRST_ROW_INDEX).build();
+        CreateExcelReportRequest request = createExcelReportRequest(DEFAULT_ROW_DATA, -1, file, TEMPLATE_DATA_FIRST_ROW_INDEX);
 
         // when
         Throwable thrown = catchThrowable(() -> excelReportService.createExcelReport(request));
 
         // then
-        assertThat(thrown).isInstanceOf(IllegalArgumentException.class);
+        assertThat(thrown).isInstanceOf(IllegalArgumentException.class).hasMessage("Batch size must be greater than zero!");
```

Comment:
Maybe introduce a custom exception instead of asserting an exception message? Not sure about this because we would probably end with a lot of custom exceptions.

Reply:
I would prefer to keep IllegalArgumentException because as you say we might end up with a lot of custom exception, and in the end this is the purpose of IllegalArgumentException